src/components/MainPage.js

```diff
@@ -0,0 +1,188 @@
+    level: "middle",
+    labels: ["frontend", "backend"],
+    image: dataStructureLogo,
+  },
+];
+const userProgress = {
+  ["userId"]: [
+    {
+      id: "react-basic",
+      attemptsNumber: "",
+      progress: 50,
+    },
+    {
+      id: "react-middle",
+      attemptsNumber: "",
+      progress: 10,
+    },
+    {
+      id: "js-basic",
+      attemptsNumber: "",
+      progress: 80,
+    },
+    {
+      id: "js-functions",
+      attemptsNumber: "",
+      progress: 100,
+    },
+  ],
+};
+
+const defaultTheme = createTheme();
+
+export default function MainPage() {
+  return (
+    <ThemeProvider theme={defaultTheme}>
+      <CssBaseline />
+      <main>
+        <Box
+          sx={{
+            pt: 6,
+          }}
+        >
+          <Container maxWidth="sm">
+            <Typography
+              component="h1"
+              variant="h4"
+              align="center"
+              color="text.primary"
+              gutterBottom
+            >
+              CHOOSE A QUIZ
+            </Typography>
+          </Container>
+        </Box>
+        <Container sx={{ py: 8 }} maxWidth="md">
+          <Grid container spacing={4}>
+            {quizCards.map((card) => (
+              <Grid item key={card.id} xs={12} sm={6} md={4}>
+                <Card
+                  sx={{
+                    height: "100%",
+                    display: "flex",
+                  }}
+                >
+                  <CardContent sx={{ display: "flex" }}>
+                    <Box direction="row" flexWrap="wrap" gap="5px">
+                      <Labels labels={card.labels}></Labels>
+                      <Chip size={"small"} label={card.level} />
+                      <Chip size={"small"} label={card.category} />{" "}
+                      <Box marginTop="80px">
+                        <CardActions>
+                          <Button variant="outlined" size="small">
+                            Start Quiz
+                          </Button>
+                        </CardActions>
+                      </Box>
+                    </Box>
+                    <Box marginLeft="50px">
+                      <Box
+                        marginTop="25px"
+                        sx={{
+                          display: "flex",
+                          flexDirection: "column",
+                          justifyContent: "center",
+                          alignItems: "center",
+                          position: "relative",
+                        }}
+                      >
+                        <CardMedia
+                          width="100%"
+                          component="div"
+                          sx={{
+                            width: "63px",
+                            height: "62px",
+                          }}
+                          image={card.image}
+                        />
+                        <Progress
+                          userProgress={userProgress}
```

Comment:
Could you try to pass on the actual user progress (maybe as a number) instead of the whole `userProgress` object?

Reply:
I'm a bit confused about this. I pass the whole object because Progress.js needs the `quizId` as well as the `quizProgress`(`userProgress`) number. If there's something I'm not seeing, please let me know and help me to understand :)

Reply:
Of course. From what I understand, `userProgress ` is not a number, but it's an object. On line 72 it's being declared:
```
const userProgress = {
  ["userId"]: [
    {
      id: "react-basic",
      attemptsNumber: "",
      progress: 50,
    },
    {
      id: "react-middle",
      attemptsNumber: "",
      progress: 10,
    },
    {
      id: "js-basic",
      attemptsNumber: "",
      progress: 80,
    },
    {
      id: "js-functions",
      attemptsNumber: "",
      progress: 100,
    },
  ],
};
```
Then you are passing the whole `userProgress` object in props:
```
<Progress
  userProgress={userProgress}
  quizId={card.id}
></Progress>
```
Then inside Progress component you are mapping through the `userProgress` and when you find the right element inside the child array, you will pass that number to the Progress component. So, my idea is that you don't need to pass the `userProgress` to the Progress component. That you actually only need the number value itself to render the Progress. For that reason, you would probably have to reorganize the data structure in a different way. It would be something like:
```
<Progress value={value} />
```

Also, is this the way you will be getting it from the backend?

Let me know what you think.